hail/src/main/scala/is/hail/expr/ir/PValue.scala

```diff
@@ -102,3 +105,36 @@ class PCanonicalIndexableValue(val pt: PContainer, val a: Code[Long]) extends PI
   def store(mb: EmitMethodBuilder, r: Code[Region], dst: Code[Long]): Code[Unit] =
     Region.storeAddress(dst, a)
 }
+
+abstract class PBaseStructValue extends PValue {
+  def pt: PBaseStruct
+
+  def isFieldMissing(fieldIdx: Int): Code[Boolean]
+
+  def loadField(fieldIdx: Int): PValue
+
+  def isFieldMissing(fieldName: String): Code[Boolean] = isFieldMissing(pt.fieldIdx(fieldName))
+
+  def isFieldDefined(fieldIdx: Int): Code[Boolean] = !isFieldMissing(fieldIdx)
+
+  def isFieldDefined(fieldName: String): Code[Boolean] = !isFieldMissing(fieldName)
+
+  def loadField(fieldName: String): PValue = loadField(pt.fieldIdx(fieldName))
```

Comment:
do we plan to introduce the same `loadField(doIfMissing, doIfDefined)` signatures here later?

Reply:
I think these two should be merged into `loadField: EmitTriplet`.  I think that answers your question?

Reply:
yeah, cool.